### PR TITLE
Enabled optional time validation in ChainTrustVaildator

### PR DESCRIPTION
**Pull request by lemccomb**
Upgrading to .NET made chain trust validation more strict about validity time on signing certificates. But some users need to allow signatures with expired certificates to pass validation.
This pull request adds an optional "allowOutdated" field to the ChainTrustValidator constructor, which allows certificates to pass chain trust validation after their "NotAfter" date unless they have a lifetime EKU.

### CODE REVIEW

File: CoseSign1.Certificates.Tests/X509ChainTrustValidatorTests.cs [hunk 1]
```diff
@@ -341,6 +343,29 @@ public void X509TrustValidatorSelfSigned()
...
+        // Validate with AllowOutdated OFF and an expired certificate
+        Validator = new(X509RevocationMode.NoCheck, false, allowUntrusted: true);
+        Validator.TryValidate(message2, out results2).Should().BeFalse();
+        results2.Count.Should().Be(1);
+        results2[0].PassedValidation.Should().BeFalse();
+        results2[0].Includes.Should().NotBeNull();
+        results2[0].Includes?.Count.Should().Be(2);
+        status = results2[0].Includes?.Cast<X509ChainStatus>()
+            .FirstOrDefault(s => s.Status == X509ChainStatusFlags.NotTimeValid);
+        status.Should().NotBeNull();
```
Comment (elantiguamsft) [inline CoseSign1.Certificates.Tests/X509ChainTrustValidatorTests.cs:368]: Great tests! Can we add the same tests for IndirectSignatureFactory()? Might make sense to copy this entire test method for the indirect version

File: CoseSign1.Certificates/Local/Validators/X509ChainTrustValidator.cs [hunk 2]
```diff
@@ -129,32 +134,44 @@ public class X509ChainTrustValidator(
...
+        flagsToIgnore |= AllowUntrusted ? X509ChainStatusFlags.UntrustedRoot : 0;
+
+        // If we have a valid user-supplied root, consider it trusted. (Not supported by .NET Standard 2.0 so we have to do it ourselves.)
+        string chainRootThumb = ChainBuilder.ChainElements.FirstOrDefault(element => element.Subject.Equals(element.Issuer))?.Thumbprint ?? string.Empty;
+        bool trustUserRoot = hasRoots && TrustUserRoots && !string.IsNullOrEmpty(chainRootThumb) && Roots!.Any(r => r.Thumbprint == chainRootThumb);
+        flagsToIgnore |= trustUserRoot ? X509ChainStatusFlags.UntrustedRoot : 0;
+
+        // If allowOutdated is set and none of the outdated certificates in the chain have a lifetime EKU, ignore NotTimeValid.
+        List<X509Certificate2>? outdatedCerts = ChainBuilder.ChainElements?.Where(c => c.NotAfter < DateTime.Now).ToList();
+        bool expired = outdatedCerts?.Any(c => c.Extensions.OfType<X509Extension>().Any(e => e.Oid?.Value == LifetimeEkuOidValue)) ?? false;
```
Comment (elantiguamsft) [inline CoseSign1.Certificates/Local/Validators/X509ChainTrustValidator.cs:154]: do we have any tests where allowOutdated is true, the cert is expired, and the certificate was created with the lifetime eku?

Comment (lemccomb) [reply CoseSign1.Certificates/Local/Validators/X509ChainTrustValidator.cs:154]: No we don't. I'll add that.

File: CoseSign1.Certificates.Tests/X509ChainTrustValidatorTests.cs (hunk 1 above)
Comment (lemccomb) [reply CoseSign1.Certificates.Tests/X509ChainTrustValidatorTests.cs:368]: I'm not sure we need it. IndirectSignatureFactory creates signatures but doesn't appear to call any validators. Is there a connection I'm not aware of?

Comment (lemccomb) [reply CoseSign1.Certificates.Tests/X509ChainTrustValidatorTests.cs]: Not a bad idea but out of scope for this PR.

File: CoseSign1.Certificates/Local/Validators/X509ChainTrustValidator.cs (hunk 2 above)
Comment (lemccomb) [reply CoseSign1.Certificates/Local/Validators/X509ChainTrustValidator.cs]: Done. And it caught a bug, so I'm glad I made the test.

**Review by elantiguamsft — APPROVED** [review]

**Pull request merged**